### ISSUE TITLE
Accuracy immediately gets stuck at 50% and stays there using either SGD or Adam

### ISSUE THREAD

**Issue by catskillsresearch** [OpenNMT/OpenNMT-py#1428]
I am trying to use Speech Demo on Swahili to English with an English corpus of 50,000 words.  I have about 80,000 short speech clips (3 or 4 seconds each), organized into batches of 64 clips each.  When I try to train this, using either Adam or SGD, it almost immediately pegs to acc: 50.00.  Any idea what is happening here?  When I test a model trained this way, it gives "<unk>" for everything.  

Here is what the preprocess looks like:

`python preprocess.py ^
-data_type audio ^
-sample_rate 8000 ^
-shard_size 64 ^
-seed 7 ^
-src_dir %stage% ^
-train_src %stage%\src_train.txt ^
-train_tgt %stage%\tgt_train.txt ^
-save_data %shr_dir%\sw
`
Here is a trace of training:

`python train.py -gpu_ranks 0 -tensorboard -report_every 1 -train_steps 3000000 -warmup_steps 1 -save_checkpoint_steps 100 -batch_size 16 -model_type audio -sample_rate 8000 -enc_rnn_size 1024 -dec_rnn_size 1024  -dropout 0 -enc_layers 6 -dec_layers 1 -audio_enc_pooling 1,1,1,1,2,2 -rnn_type LSTM -global_attention mlp -max_grad_norm 100 -optim sgd -learning_rate .01 -learning_rate_decay 0.9 -data e:\CRC_STAGE\BUILD\audio\shards\sw -save_model e:\CRC_STAGE\BUILD\audio\model\model_1024_1024_6_1_LSTM_sgd 
[2019-05-09 07:50:57,152 INFO]  * tgt vocab size = 50004
[2019-05-09 07:50:57,152 INFO] Building model...
[2019-05-09 07:51:01,480 INFO] NMTModel(
  (encoder): AudioEncoder(
    (W): Linear(in_features=1024, out_features=1024, bias=False)
    (batchnorm_0): BatchNorm1d(1024, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (rnn_0): LSTM(81, 1024)
    (pool_0): MaxPool1d(kernel_size=1, stride=1, padding=0, dilation=1, ceil_mode=False)
    (rnn_1): LSTM(1024, 1024)
    (pool_1): MaxPool1d(kernel_size=1, stride=1, padding=0, dilation=1, ceil_mode=False)
    (batchnorm_1): BatchNorm1d(1024, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (rnn_2): LSTM(1024, 1024)
    (pool_2): MaxPool1d(kernel_size=1, stride=1, padding=0, dilation=1, ceil_mode=False)
    (batchnorm_2): BatchNorm1d(1024, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (rnn_3): LSTM(1024, 1024)
    (pool_3): MaxPool1d(kernel_size=1, stride=1, padding=0, dilation=1, ceil_mode=False)
    (batchnorm_3): BatchNorm1d(1024, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (rnn_4): LSTM(1024, 1024)
    (pool_4): MaxPool1d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
    (batchnorm_4): BatchNorm1d(1024, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (rnn_5): LSTM(1024, 1024)
    (pool_5): MaxPool1d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
    (batchnorm_5): BatchNorm1d(1024, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
  )
  (decoder): InputFeedRNNDecoder(
    (embeddings): Embeddings(
      (make_embedding): Sequential(
        (emb_luts): Elementwise(
          (0): Embedding(50004, 500, padding_idx=1)
        )
      )
    )
    (dropout): Dropout(p=0.0)
    (rnn): StackedLSTM(
      (dropout): Dropout(p=0.0)
      (layers): ModuleList(
        (0): LSTMCell(1524, 1024)
      )
    )
    (attn): GlobalAttention(
      (linear_context): Linear(in_features=1024, out_features=1024, bias=False)
      (linear_query): Linear(in_features=1024, out_features=1024, bias=True)
      (v): Linear(in_features=1024, out_features=1, bias=False)
      (linear_out): Linear(in_features=2048, out_features=1024, bias=True)
    )
  )
  (generator): Sequential(
    (0): Linear(in_features=1024, out_features=50004, bias=True)
    (1): Cast()
    (2): LogSoftmax()
  )
)
[2019-05-09 07:51:01,480 INFO] encoder: 47579136
[2019-05-09 07:51:01,480 INFO] decoder: 90898276
[2019-05-09 07:51:01,480 INFO] * number of parameters: 138477412
[2019-05-09 07:51:01,574 INFO] Starting training on GPU: [0]
[2019-05-09 07:51:01,574 INFO] Start training loop without validation...
[2019-05-09 07:51:01,574 INFO] Loading dataset from e:\CRC_STAGE\BUILD\audio\shards\sw.train.0.pt, number of examples: 64
[2019-05-09 07:51:02,948 INFO] Step  1/3000000; acc:   0.00; ppl: 62608.59; xent: 11.04; lr: 0.01000; 11283/ 23 tok/s;      1 sec
[2019-05-09 07:51:03,276 INFO] Step  2/3000000; acc:  50.00; ppl: 535.92; xent: 6.28; lr: 0.01000; 9554/ 98 tok/s;      2 sec
[2019-05-09 07:51:04,479 INFO] Step  3/3000000; acc:  50.00; ppl: 1531.69; xent: 7.33; lr: 0.01000; 15354/ 27 tok/s;      3 sec
[2019-05-09 07:51:05,104 INFO] Step  4/3000000; acc:   9.38; ppl: 12203.64; xent: 9.41; lr: 0.01000; 11250/ 51 tok/s;      4 sec
[2019-05-09 07:51:05,120 INFO] Loading dataset from e:\CRC_STAGE\BUILD\audio\shards\sw.train.1.pt, number of examples: 64
[2019-05-09 07:51:05,494 INFO] Step  5/3000000; acc:  50.00; ppl: 294.10; xent: 5.68; lr: 0.01000; 9845/ 82 tok/s;      4 sec
[2019-05-09 07:51:05,619 INFO] Step  6/3000000; acc:  50.00; ppl: 263.75; xent: 5.57; lr: 0.01000; 7833/256 tok/s;      4 sec
[2019-05-09 07:51:06,682 INFO] Step  7/3000000; acc:  50.00; ppl: 367.47; xent: 5.91; lr: 0.01000; 8164/ 30 tok/s;      5 sec
[2019-05-09 07:51:06,869 INFO] Step  8/3000000; acc:  50.00; ppl: 238.48; xent: 5.47; lr: 0.01000; 9836/171 tok/s;      5 sec
[2019-05-09 07:51:06,885 INFO] Loading dataset from e:\CRC_STAGE\BUILD\audio\shards\sw.train.10.pt, number of examples: 64
[2019-05-09 07:51:07,213 INFO] Step  9/3000000; acc:  50.00; ppl: 254.72; xent: 5.54; lr: 0.01000; 10050/ 93 tok/s;      6 sec
[2019-05-09 07:51:07,338 INFO] Step 10/3000000; acc:  50.00; ppl: 248.83; xent: 5.52; lr: 0.01000; 9144/293 tok/s;      6 sec
[2019-05-09 07:51:08,228 INFO] Step 11/3000000; acc:  50.00; ppl: 200.26; xent: 5.30; lr: 0.01000; 7940/ 36 tok/s;      7 sec
[2019-05-09 07:51:08,416 INFO] Step 12/3000000; acc:  50.00; ppl: 283.40; xent: 5.65; lr: 0.01000; 9529/171 tok/s;      7 sec
`

**Comment by catskillsresearch** [comment]
Oh never mind, this was a stupid mistake.  The target file is supposed to contain the text of the audio, one line per clip.  I was putting the name of a file containing the audio for that clip.  So it was getting a sequence of almost identical file names 50% right.
